src/System.Net.Sockets/tests/FunctionalTests/DualModeSocketTest.cs

```diff
@@ -436,37 +416,23 @@ public void Socket_BeginConnectV4IPEndPointToV4Host_Throws()
         }
 
         [Fact]
-        public void BeginConnectV4IPEndPointToV4Host_Success()
-        {
-            DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.Loopback, IPAddress.Loopback, false);
-        }
+        public Task BeginConnectV4IPEndPointToV4Host_Success() => DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.Loopback, IPAddress.Loopback, false);
 
         [Fact]
-        public void BeginConnectV6IPEndPointToV6Host_Success()
-        {
-            DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Loopback, false);
-        }
+        public Task BeginConnectV6IPEndPointToV6Host_Success() => DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Loopback, false);
 
         [Fact]
-        public void BeginConnectV4IPEndPointToDualHost_Success()
-        {
-            DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.Loopback, IPAddress.IPv6Any, true);
-        }
+        public Task BeginConnectV4IPEndPointToDualHost_Success() => DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.Loopback, IPAddress.IPv6Any, true);
 
         [Fact]
-        public void BeginConnectV6IPEndPointToDualHost_Success()
-        {
-            DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Any, true);
-        }
+        public Task BeginConnectV6IPEndPointToDualHost_Success() => DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress.IPv6Loopback, IPAddress.IPv6Any, true);
 
-        private void DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, bool dualModeServer)
+        private async Task DualModeBeginConnect_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, bool dualModeServer)
         {
-            int port;
             using (Socket socket = new Socket(SocketType.Stream, ProtocolType.Tcp))
-            using (SocketServer server = new SocketServer(_log, listenOn, dualModeServer, out port))
+            using (SocketServer server = new SocketServer(_log, listenOn, dualModeServer, out int port))
             {
-                IAsyncResult async = socket.BeginConnect(new IPEndPoint(connectTo, port), null, null);
-                socket.EndConnect(async);
```

Comment:
In removing this blocking, do we still have coverage of said blocking?  We shouldn't remove all of it: we still need to test the path where EndXxx is called in this pattern, blocking waiting for BeginConnect to complete when no callback is provided.

Reply:
That is true, we don't get hard blocking tests by removing them. On the flipside previously there was no callback testing :) I could add a single blocking test to cover just the blocking functionality and leave the rest as call backs?

Reply:
Ok